src/py/flwr/common/grpc.py

```diff
@@ -38,14 +39,14 @@ def create_channel(
         ("grpc.max_receive_message_length", max_message_length),
     ]
 
-    if root_certificates is not None:
+    if insecure:
+        channel = grpc.insecure_channel(server_address, options=channel_options)
+        log(INFO, "Opened insecure gRPC connection (no certificates were passed)")
+    else:
         ssl_channel_credentials = grpc.ssl_channel_credentials(root_certificates)
         channel = grpc.secure_channel(
             server_address, ssl_channel_credentials, options=channel_options
         )
         log(INFO, "Opened secure gRPC connection using certificates")
```

Comment:
Should we throw an exception if `insecure` is `True` and `root_certificates` are passed along?

Reply:
Updated.